client/blocks/upload-image/index.jsx

```diff
@@ -226,14 +229,18 @@ class UploadImage extends Component {
 
 		const classes = classnames( 'upload-image-modal', additionalImageEditorClasses );
 
+		const isEditingDefaultImage = defaultImage && selectedImage === defaultImage.URL;
+
+		const media = isEditingDefaultImage ? defaultImage : {
+			src: selectedImage,
```

Comment:
`defaultImage` is a media item with `URL` property. `ImageEditor` accepts either `URL` or `src` props.

`defaultImage` as documented is either a media item or its id. ID was allowed as value as a convenience because getting media item was a hassle and didn't see it readily available from calling side.

I did have both `defaultImageId` and `defaultImage` props in the first commit but opted to combine two values. I can see your point re use of `PropTypes.any` so let's revisit this when mixed props confuse users. I am more used to TypeScript which allows expressions like `MediaItem | number | undefined` as React property type.